src/meta/upgrade/MetaDataUpgrade.cpp

```diff
@@ -434,5 +563,10 @@ void MetaDataUpgrade::printJobDesc(const folly::StringPiece &key, const folly::S
   LOG(INFO) << "JobDesc stopTime: " << stopTime;
 }
 
+Status MetaDataUpgrade::saveMachineAndZone(std::vector<kvstore::KV> data) {
```

Comment:
extra copy of data?

Reply:
not sure Return Value Optimization maybe fix it.